File: Phanfare.ExternalAPI/NewsItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;

namespace Phanfare.ExternalAPI
{
	public enum NewsItemType
	{
		NewAlbum,
		NewMedia,
	}

	public class NewsItem
	{
		public NewsItemType Type { get; set; }
		public DateTime Timestamp { get; set; }
		public long SourceID { get; set; }
		public string SourceName { get; set; }
		public string SiteName { get; set; }
		public long AlbumID { get; set; }
		public string AlbumName { get; set; }
		public string AlbumURL { get; set; }
		public string AlbumSlideshowURL { get; set; }

		public ImageInfo[] LeadImages { get; set; }

		public static NewsItem FromXML( XmlElement el )
		{
			NewsItem item = new NewsItem();
			item.Type = el.GetAttributeEnum<NewsItemType>( "type" );
			item.Timestamp = el.GetAttributeDate( "timestamp" );
			item.SourceID = el.GetAttributeInt64( "source_uid" );
			item.SourceName = el.GetAttribute( "source_name" );
			item.SiteName = el.GetAttribute( "site_name" );
			item.AlbumID = el.GetAttributeInt64( "album_id" );
			item.AlbumName = el.GetAttribute( "album_name" );
			item.AlbumURL = el.GetAttribute( "album_url" );
			item.AlbumSlideshowURL = el.GetAttribute( "album_slideshow_url" );

			XmlElement imagesElement = el[ "images" ];
			item.LeadImages = imagesElement.ReadList<ImageInfo>( "num_images", ImageInfo.FromXML );

			return item;
		}
	}
}

File: trunk/PhanfareJavaAPI/src/com/phanfare/api/NewsItem.java
package com.phanfare.api;

import java.util.Date;

import org.json.me.JSONArray;
import org.json.me.JSONException;
import org.json.me.JSONObject;

public class NewsItem {
	public final static int NEWS_TYPE_NEW_ALBUM = 0;
	public final static int NEWS_TYPE_NEW_MEDIA = 1;

	public int type;
	public Date timestamp;
	public long sourceId;
	public String sourceName;
	public String siteName;
	public long albumId;
	public String albumName;
	public String albumUrl;
	public String albumSlideshowUrl;

	public ImageInfo[] leadImages;

	public static NewsItem fromJson(JSONObject object) {
		NewsItem item = new NewsItem();
		try {
			String type = object.getString("type").toLowerCase();
			if (type.equals("newalbum") == true)
				item.type = NewsItem.NEWS_TYPE_NEW_ALBUM;
			else
				item.type = NewsItem.NEWS_TYPE_NEW_MEDIA;
			item.timestamp = Utilities.convertJsonDate(object.getString("timestamp"));
			item.sourceId = object.getLong("source_uid");
			item.sourceName = object.getString("source_name");
			item.siteName = object.getString("site_name");
			item.albumId = object.getLong("album_id");
			item.albumName = object.getString("album_name");
			item.albumUrl = object.getString("album_url");
			item.albumSlideshowUrl = object.getString("album_slideshow_url");
			JSONArray images = object.getJSONArray("images");
			item.leadImages = new ImageInfo[images.length()];
			for (int n = 0; n < item.leadImages.length; n++) {
				item.leadImages[n] = ImageInfo.fromJson(images.getJSONObject(n));
			}

		} catch (JSONException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return null;
		}
		return item;
	}
}
